### ISSUE TITLE
Migration back to gh-pages

### ISSUE THREAD

**Issue by rossabaker** [http4s/http4s-jdk-http-client#563]
The main site is down, and nobody else can see Netlify, so I migrated back to gh-pages so I'm not the single point of failure.

I've set up a CNAME, but GitHub still won't accept it:

```console
> dig jdk-http-client.http4s.org +noall +answer -t A

; <<>> DiG 9.10.6 <<>> jdk-http-client.http4s.org +noall +answer -t A
;; global options: +cmd
jdk-http-client.http4s.org. 2894 IN	CNAME	http4s.github.io.
http4s.github.io.	2894	IN	A	185.199.108.153
http4s.github.io.	2894	IN	A	185.199.109.153
http4s.github.io.	2894	IN	A	185.199.110.153
http4s.github.io.	2894	IN	A	185.199.111.153
```

The `/stable` and `/latest` redirects will no longer work.  I see a `/latest` directory in gh-pages.

If I ignore the security warning (which I expect will resolve itself), I get a 404.  We might need an index.html or some other landing page that redirects to a particular version.

**Comment by amesgen** [comment]
I am working on a transition to Leika as part of https://github.com/http4s/http4s-jdk-http-client/pull/562, and Laika supports versioned documentation, so the need for redirects might go away as a side effect.

**Comment by rossabaker** [comment]
Sorry to dump this on you abruptly, but everything's already broken, and I knew you were working on a refurb.  I'll keep an eye on things, and if you need DNS help or don't have enough GitHub rights, please yell.

**Comment by armanbilge** [comment]
https://jdk-http-client.http4s.org/ is back :) still needs some follow-up, to link to the previous versions.

**Comment by armanbilge** [comment]
Things mostly looking up at https://jdk-http-client.http4s.org :)

The one big surprise was that the 1.x site encourages you to install 0.3.4! I think this is because you are using squash merges rather than history-preserving merges.

**Comment by amesgen** [comment]
Hmm, how can this influence which git tag is considered to be the latest one?

**Comment by armanbilge** [comment]
It's because the history on main doesn't include any tags between `0.3.4` and `0.6.0-M?`.

For example, if we look https://github.com/http4s/http4s-jdk-http-client/commit/98082981e908cb31ff38aed3f60b73f97e1e9e3a, which is the commit tagged `v0.5.0` we see that it exists only on the series/0.5 and series/0.7 branches. It was "squashed" into main, so it lost the true history.

**Comment by amesgen** [comment]
Ah, that makes sense, I hadn't thought of that downside of just cherry-picking commits from `main` to the older branches like I did, properly merging everything would have prevented this 😭 . But with #586, more useful versions are now displayed either way, thanks!

**Comment by amesgen** [comment]
I think that the last remaining deficiency of the new site setup was fixed by #603 (the version in the sbt snippet on https://jdk-http-client.http4s.org/1.x is now correct), thanks @armanbilge!
